trin-core/src/types/accumulator.rs

```diff
@@ -44,6 +45,59 @@ impl MasterAccumulator {
         }
     }
 
+    pub fn latest_height(&self) -> i64 {
+        let historical_height = self.epoch_number() * EPOCH_SIZE as i64;
+        historical_height + self.current_epoch.header_records.len() as i64
+    }
+
+    pub fn epoch_number(&self) -> i64 {
+        self.historical_epochs.epochs.len() as i64
+    }
+
+    /// Compare that one master accumulator contains the same validation data as another,
+    /// regardless of each accumulator's current height.
+    pub fn compare(&self, other: &Self) -> anyhow::Result<()> {
+        let historical_epochs = self
+            .historical_epochs
+            .epochs
+            .into_iter()
+            .zip(other.historical_epochs.epochs.into_iter());
+        for (index, (self_epoch, other_epoch)) in historical_epochs.into_iter().enumerate() {
+            if self_epoch != other_epoch {
+                return Err(anyhow!(
+                    "Accumulator mismatch at historical epoch #{:?}.
+                    Expected historical epoch root: {:?}.
+                    Found historical epoch root: {:?}",
+                    index,
+                    self_epoch,
+                    other_epoch
+                ));
+            }
+        }
+        // only check header records if accs. in same epoch.
+        if self.epoch_number() != other.epoch_number() {
+            return Ok(());
+        }
+        let header_records = self
+            .current_epoch
+            .header_records
+            .into_iter()
+            .zip(other.current_epoch.header_records.into_iter());
+        for (index, (self_hr, other_hr)) in header_records.into_iter().enumerate() {
+            if self_hr != other_hr {
+                return Err(anyhow!(
+                    "Accumulator mismatch at header record #{:?}.
+                    Expected header record: {:?}.
+                    Found header record: {:?}",
+                    index,
+                    self_hr,
+                    other_hr
+                ));
+            }
+        }
```

Comment:
We should be able to simplify this quite a lot by comparing common prefix slices of the historical epochs and/or header records in the latest common epoch directly.

Reply:
I guess the design decision here - comparing header by header instead of comparing a slice of headers, is to know at which header record there is a mismatch?